### PR TITLE
fix for rsync protocol

### PR DESCRIPTION
**Pull request by hyeonseongc**
##### ISSUE TYPE

- Bugfix Pull Request
##### COMPONENT NAME

files/synchronize.py
##### ANSIBLE VERSION

```
2.1.0.0
```
##### SUMMARY

I've requested this feature in ansible/ansible-modules-core#1999 and it's merged
but after I update ansible to 2.1, this feature is not working as I expected.
here's patch for fix this issue.
sorry for inconvenient.

work with ansible/ansible#16756

```

```

### CODE REVIEW

**Comment by gregdek** [comment]
Thanks @peter-jang. To the current maintainers, @tima please review according to guidelines (http://docs.ansible.com/ansible/developing_modules.html#module-checklist) and comment with text 'shipit', 'needs_revision' or 'close_me' as appropriate.

[This message brought to you by your friendly Ansibull-bot.]

File: files/synchronize.py [hunk 1]
```diff
@@ -407,10 +407,10 @@ def main():
     if ssh_args:
       ssh_opts = '%s %s' % (ssh_opts, ssh_args)
 
-    if source.startswith('rsync://') and dest.startswith('rsync://'):
+    if source.startswith('"rsync://') and dest.startswith('"rsync://'):
```
Comment (tima) [inline files/synchronize.py:410]: I find these two a bit curious. Could you explain what you found that the " character is required here?

**Comment by tima** [comment]
waiting_on_contributor

**Comment by tima** [comment]
needs_info

**Comment by gregdek** [comment]
@peter-jang, a community member requested some information about this PR; for that reason, we put this PR into the 'needs_info' state.

When you have answered the questions, please comment with text 'ready_for_review' and we will put this PR back into review.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by hyeonseongc** [comment]
@tima, this is my fault.
from the first part of this module, after start main function (Line 320)
source and dest parameters are retrieved like this

`source = '"' + module.params['src'] + '"'`
`dest = '"' + module.params['dest'] + '"'`

because of this, src and dest path start with " character.
but my first PR, I've tested this feature like 

`if 'rsync://' in source`

after you mentioned this code should be source.startwith() ,
I've changed but didn't test with new code..

before my PR merged into main branch, I still used my old code
and I just figured this out just few weeks ago.

let me know if there's still issue.

ready_for_review

**Comment by gregdek** [comment]
Thanks @peter-jang. To the current maintainers, @tima please review according to guidelines (http://docs.ansible.com/ansible/developing_modules.html#module-checklist) and comment with text 'shipit', 'needs_revision' or 'close_me' as appropriate.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by tima** [comment]
@peter-jang ah yes, missed those lines. it was a safety measure in case the src and/or dest had something that needed to be quoted we're quoting everything that comes thru. 

shipit

**Comment by abadger** [comment]
Also cherry-picked to the stable-2.1 branch

**Comment by tima** [comment]
@abadger -- this is only half of the fix. There is an unmerged core PR #16756 that goes with this. That one is held up waiting for tests.

**Pull request merged**